src/worker.cpp

```diff
@@ -1185,4 +1184,281 @@ void Worker::collectFrusFromJson()
         }
     }
 }
+
+std::string Worker::getPrimaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("PrimaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+std::string Worker::getSecondaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("SecondaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+types::BackupMapVariant Worker::getBackupMap()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        const std::string& l_backupMap =
+            m_parsedJson["backupAndRestore"].value("BackupMap", "");
+        if (l_backupMap.find("IBMSystemBackupMap") != std::string::npos)
```

Comment:
This does not align with the design discussed.
Why the if condition has the hard coded value?
If name of the map is changed or a new map name is introduced, the code needs to be modified here with a new if or else case.
The selection of map should be at runtime based on the value read from JSON. API should not be modified each time there is a new addition.